homeassistant/components/fordpass/config_flow.py

```diff
@@ -0,0 +1,76 @@
+"""Config flow for FordPass integration."""
+import logging
+
+from fordpass import Vehicle
+import voluptuous as vol
+
+from homeassistant import config_entries, core, exceptions
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+
+from .const import DOMAIN, VIN  # pylint:disable=unused-import
+
+_LOGGER = logging.getLogger(__name__)
+
+DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_USERNAME): str,
+        vol.Required(CONF_PASSWORD): str,
+        vol.Required(VIN): vol.All(str, vol.Length(min=17, max=17)),
+    }
+)
+
+
+async def validate_input(hass: core.HomeAssistant, data):
+    """Validate the user input allows us to connect.
+
+    Data has the keys from DATA_SCHEMA with values provided by the user.
+    """
+    vehicle = Vehicle(data[CONF_USERNAME], data[CONF_PASSWORD], data[VIN])
+
+    try:
+        result = await hass.async_add_executor_job(vehicle.auth)
+
+    except Exception as ex:
```

Comment:
My impression is that is is preferred to have more narrow exceptions, based on comments i've seen for other new integrations.  Perhaps make a library specific exception?

(Applies here and also in __init__.py)

Reply:
It's not required but a library-specific exception is preferred.


`Exception` is too broad to raise InvalidAuth can it be more specific?